File: src/command/ActivityUpdate.ts
/*
 * Copyright 2021-2022 Golden_Water
 * SPDX-License-Identifier: AGPL-3.0-only
 */

export type ActivityUpdateCmd = "activityUpdate";

export type ActivityUpdate = {
  cmd: ActivityUpdateCmd;
  activity: number;
};

export function getActivityUpdateMessage(activity: number): ActivityUpdate {
  return {
    cmd: "activityUpdate",
    activity: activity,
  };
}

export function getActivityUpdateMessageCmd(): ActivityUpdateCmd {
  return "activityUpdate";
}

File: share/src/command_packet/bilibili_command/activity_update.rs
/*
 * Copyright 2021-2023 Golden_Water
 * SPDX-License-Identifier: AGPL-3.0-only
 */

#[derive(serde::Serialize, serde::Deserialize, PartialEq, Eq, Debug, Clone)]
#[serde(rename_all = "camelCase")]
pub struct ActivityUpdate {
  activity: u32,
}

impl ActivityUpdate {
  pub fn new(activity: u32) -> ActivityUpdate {
    ActivityUpdate { activity }
  }
}
